File: flink-demo/flinkScalaCode/src/main/scala/com/yw/flink/example/scalacases/case02_transformation/Case07_Union.scala
package com.yw.flink.example.scalacases.case02_transformation

import org.apache.flink.streaming.api.scala.StreamExecutionEnvironment

/**
  *
  * @author yangwei
  */
object Case07_Union {
  def main(args: Array[String]): Unit = {
    val env: StreamExecutionEnvironment = StreamExecutionEnvironment.getExecutionEnvironment
    import org.apache.flink.streaming.api.scala._
    val ds1: DataStream[Int] = env.fromCollection(List(1, 2, 3, 4, 5))
    val ds2: DataStream[Int] = env.fromCollection(List(5, 4, 8, 9, 10))
    val result: DataStream[Int] = ds1.union(ds2)
    result.print()
    env.execute()
  }
}

File: flink-demo/flinkJavaCode/src/main/java/com/yw/flink/example/javacases/case02_transformation/Case07_Union.java
package com.yw.flink.example.javacases.case02_transformation;

import org.apache.flink.streaming.api.datastream.DataStream;
import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

import java.util.Arrays;

public class Case07_Union {
    public static void main(String[] args) throws Exception {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        DataStreamSource<Integer> ds1 = env.fromCollection(Arrays.asList(1, 2, 3, 4, 5));
        DataStreamSource<Integer> ds2 = env.fromCollection(Arrays.asList(6,7,8,9,10));

        //union合并
        DataStream<Integer> union = ds1.union(ds2);
        union.print();
        env.execute();
    }
}
